packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.tsx

```diff
@@ -1,42 +1,47 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import invariant from 'tiny-invariant';
+import { TPermissions, TPermissionName } from '../../types';
 import isNil from 'lodash/isNil';
 import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
 import Authorized from '../authorized';
 
-const getHasChildren = children => React.Children.count(children) > 0;
+const getHasChildren = (children: React.ReactNode) =>
+  React.Children.count(children) > 0;
 
-export class RestrictedByPermissions extends React.Component {
-  static displayName = 'RestrictedByPermissions';
-  static propTypes = {
-    shouldMatchSomePermissions: PropTypes.bool,
-    permissions: PropTypes.arrayOf(PropTypes.string.isRequired),
-
-    unauthorizedComponent(props, propName, componentName, ...rest) {
-      if (!isNil(props[propName]) && typeof props.children === 'function') {
-        return new Error(
-          `Invalid prop \`${propName}\` supplied to \`${componentName}\`. Do not pass \`children\` as a function and \`unauthorizedComponent\`.`
-        );
-      }
-
-      return PropTypes.func(props, propName, componentName, ...rest);
-    },
-    children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
-    render: PropTypes.func,
-
-    // Injected
-    applicationContext: PropTypes.shape({
-      permissions: PropTypes.objectOf(PropTypes.bool).isRequired,
-    }).isRequired,
+type TRenderProp = (props: { isAuthorized: boolean }) => React.ReactNode;
+type Props = {
+  shouldMatchSomePermissions?: boolean;
+  permissions: TPermissionName[];
+  unauthorizedComponent?: React.ComponentType;
+  children: React.ReactNode | TRenderProp;
+  render: TRenderProp;
```

Comment:
I just went with the same prefixing. Feels quite good to tell types from other things.

Reply:
Nice, should we then have this as a convention and always prefix them with `T`?

Reply:
I think so right now. Only in case of overlap or when you feel fizzy would be odd. I wouldn‘t like prefixing interfaces with I cause they‘re also just types. So all just T would get my vore.